Core/arch/linux/FX3handler.cpp

```diff
@@ -23,35 +25,85 @@ bool fx3handler::Open(uint8_t* fw_data, uint32_t fw_size)
 
 bool fx3handler::Control(FX3Command command, uint8_t data)
 {
-    return usb_device_control(this->dev, command, 0, 0, (uint8_t *) &data, sizeof(data), 0) == 0;
+    if(stream) {
+        // pause streaming for fast control execution
+        streaming_stop(stream);
+        streaming_reset_status(stream);
+    }
+    bool ret = usb_device_control(this->dev, command, 0, 0, (uint8_t *) &data, sizeof(data), 0) == 0;
+    if(stream) {
+        // resume streaming
+        streaming_start(stream);
+    }
```

Comment:
Here I was trying to avoid freezes when using device controls during active streaming. Now it freezes less, but not perfect, if anyone can suggest a better way to fix it I'll be glad.

Reply:
I looked at libusb code once. I am not expert. It is my suspicion that we didn't handle multi thread well. I suggest this page:
http://libusb.sourceforge.net/api-1.0/libusb_mtasync.html

Reply:
Oh yes, it looks like we have a situation described at the end of the article on your link.  Thanks, I'll try to figure it out.

Reply:
I tried to figure out what happens in bulk transfers with libusb, but with no success. I found that streaming only works if there is a control transfers to the device (such as reading a debug trace), but under such conditions, all other controls are delayed by about a second. Without the control transfers bulk transfers timeouts, it looks like it tries to work when setting the transfer timeout to lowest possible 1ms, but this is not enough for full functionality anyway.
My competence is not enough to resolve this issue.

Reply:
Looks like I near solved this. Now it's possible to normally use streaming on linux, but sometimes streaming stops from libusb timeouts.
Also I was forced to reduce QUEUE_SIZE to 16 from 32, because with 32 I wasn't able get stable streaming with libusb.
Unfortunately, I can not test is something broke on Windows or not.

Reply:
Pavel thanks for the code. It compiles in windows. In release at first look it seems to run ok while in debug it stops streaming after start up.
We have to understand what's happening  before merging it to master.
Howard in the while is making big change in #199 and #200.

Reply:
Hm, it's looking as in debug we have enabled reading debug from fx3 and this not works...
It's strange that without my change in firmware, reading not working good on Linux...
Anyway debug read on linux currently not working because of timeouts in streaming so I will just revert this change.

Reply:
I reverted changes in firmware, please check when you will have a time.